DQM/SiPixelMonitorDigi/src/SiPixelDigiSource.cc

```diff
@@ -101,35 +103,94 @@ SiPixelDigiSource::~SiPixelDigiSource() {
   LogInfo("PixelDQM") << "SiPixelDigiSource::~SiPixelDigiSource: Destructor" << endl;
 }
 
-std::shared_ptr<bool> SiPixelDigiSource::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
-                                                                    const edm::EventSetup& iSetup) const {
+std::shared_ptr<SiPixelDigiCounter> SiPixelDigiSource::globalBeginLuminosityBlock(const edm::LuminosityBlock& lumi,
+                                                                                  const edm::EventSetup& iSetup) const {
   unsigned int currentLS = lumi.id().luminosityBlock();
   bool resetCounters = (currentLS % 10 == 0) ? true : false;
-  return std::make_shared<bool>(resetCounters);
+  auto lsc = std::make_shared<SiPixelDigiCounter>();
```

Comment:
I suspect this is not what you want. Now a new `SiPixelDigiCounter` is created for every lumi, and in principle the "reset" logic below is unnecessary (except it sets the booleans to `true` while the constructor sets them to `false`).

What I tried to suggest in https://github.com/cms-sw/cmssw/pull/32805#issuecomment-773473223 would be to have a member variable e.g. `std::shared_ptr<SiPixelDigiCounter> latestCounters_`, and then here
```cpp
if (not modOn or resetCounters) {
  latestCounters_ = std::make_shared<SiPixelDigiCounter>();
}
```
and have all the resetting logic in the constructor of `SiPixelDigiCounter`.